core/src/lib/cram_md5.cc

```diff
@@ -167,6 +190,15 @@ bool CramMd5Handshake::CramMd5Response()
     }
   }
 
+  auto result = CompareChallengeWithOwnQualifiedName(chal.c_str());
+
+  if (result == ComparisonResult::IS_SAME ||
```

Comment:
I would suggest to log a warning when IS_SAME ist true like "Warning: detected attack:  I will not answer to my own challenge. Please check integrity of <name of client that wants to connnect>"